### PR TITLE
Add backticks to the gramar

### LINKED ISSUE #19: Add `backtick` handling for field names and attributes
Part of #15 

Update the grammar to accept \`backtick\` fields. Dots between the backticks will be interpreted as part of the field name, not as an additional attribute.

### PR DESCRIPTION
**Pull request by rw-access**
## Issues
Relates to #15 
Closes #19 

## Details
Fairly straightforward. Added \`backtick\` support to the grammar and parser.
Validated that the engine is still working as expected.

### CODE REVIEW

File: eql/etc/eql.g [hunk 1]
```diff
@@ -75,19 +75,31 @@ string: DQ_STRING
...
+
+
 LCASE_LETTER: "a".."z"
 UCASE_LETTER: "A".."Z"
 DIGIT: "0".."9"
 
 LETTER: UCASE_LETTER | LCASE_LETTER
 WORD: LETTER+
 
+ESCAPED_NAME: "`" /[^`]+/ "`"
```
Comment (brokensound77) [inline eql/etc/eql.g:102]: whitespace allowed within the backticks?

Comment (brokensound77) [reply eql/etc/eql.g:102]: If so, can you add a test with an escaped field with whitespace

Comment (rw-access) [reply eql/etc/eql.g]: e9138e7

File: tests/test_parser.py [hunk 2]
```diff
@@ -301,6 +302,38 @@ def test_invalid_queries(self):
...
+            # now render back as text and parse again
+            node2 = parse_expression(node.render())
+            self.assertEqual(node2, node)
+
+        parse_to("`foo-bar-baz`", ["foo-bar-baz"])
+        parse_to("`foo bar baz`", ["foo bar baz"])
+        parse_to("`foo.bar.baz`", ["foo.bar.baz"])
+        parse_to("`foo`.`bar-baz`", ["foo", "bar-baz"])
+        parse_to("`foo`.`bar-baz`", ["foo", "bar-baz"])
+        parse_to("`💩`", ["💩"])
```
Comment (aleksmaus) [inline tests/test_parser.py:321]: 💩

File: tests/test_parser.py [hunk 3]
```diff
@@ -301,6 +302,38 @@ def test_invalid_queries(self):
...
+
+            # now render back as text and parse again
+            node2 = parse_expression(node.render())
+            self.assertEqual(node2, node)
+
+        parse_to("`foo-bar-baz`", ["foo-bar-baz"])
+        parse_to("`foo bar baz`", ["foo bar baz"])
+        parse_to("`foo.bar.baz`", ["foo.bar.baz"])
+        parse_to("`foo`.`bar-baz`", ["foo", "bar-baz"])
+        parse_to("`foo`.`bar-baz`", ["foo", "bar-baz"])
```
Comment (aleksmaus) [inline tests/test_parser.py:320]: two lines above are the same. did you mean something else for the second one?
